flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -374,7 +374,7 @@ class FMCMainDisplay extends BaseAirliners {
             }
 
             case FmgcFlightPhases.CRUISE: {
-                if (this.canShowNextPerfPage(_lastFlightPhase)) {
+                if (this.canShowNextPerfPage(_lastFlightPhase) || this.canShowNextPerfPage(_curFlightPhase)) {
```

Comment:
Are we sure about this? If one MCDU re-establishes CRZ from APPR (by re-entering a CRZ flight level) are we sure the other one should leave APPR to go to CRZ? APPR is still perfectly valid, as it's a future flight phase.

Reply:
@beheh this has nothing todo with that, it only ensures to switch to the next flight phases page in order or reload the the one for the current page if already selected.

Reply:
> it only ensures to switch to the next flight phases page in order or reload the the one for the current page if already selected.

No, that's exactly what I mean though. Doesn't this line mean that if your MCDU has APPR open, and you reengage CRZ through some other way (i.e. the other MCDU), we switch the page back to CRZ? This switch generally makes sense when going "forwards" through the flight phases, for example of course when we have the TAKE OFF page open and we go to CLB, we want to switch the page to CLB (TAKE OFF even becomes unreachable). But if we have APPR open, which is *after* CRZ, and somehow CRZ reengages, should the really page go "backwards" to reach CRZ? It seems perfectly valid to have APPR open during either CRZ and APPR, so I'm not sure it should automatically go backwards.

Reply:
I still don't understand the issue, when you even if you change flight phases in an oddly order why shouldn't the performance page allow you to go to that flight phase again?

Either way, when in APPR and going into CRZ, the page won't automatically transition onto that page.
It only goes to that page when the page was a) the page one before, or b) the page was already open and is the corresponding page to the new flight phase.

Basically:
- CLB -> CRZ (while on perf CLB page): perf page auto transition;
 - CLB -> CRZ (while on perf CRZ page)): crz perf page auto reload;
 - CLB -> CRZ (while on perf DES page): no autotransition or auto reload;
 - APPR -> CRZ (while on perf APPR page): perf page auto transition:
 - APPR -> CRZ (while on GO AROUND page): no autotransition or auto reload;

Reply:
> APPR -> CRZ (while on perf APPR page): perf page auto transition:

This is the behavior I'm questioning, and whether you're sure about it. The autotransition going forward makes sense, because the old page becomes unreachable, however when you transition to CRZ, APPR stays continuously available. Why should it auto-transition away from APPR in this case?

Reply:
Now I understand, I assumed that the performance page (when displaying the current page relative to flight phase) will transition to the page corresponding to the new flight phase.

Reply:
I'll have to check if I can find something about this behavior in our references, but I highly doubt that.

Reply:
okay I wasn't able to find anything, BUT it's impossible for the fmgc to automatically transition from APPROACH to a previous flight phase, which means the issue is no longer relevant as it only covers automatic flight phase transitions.

Manual transitions from APPR to CLB or CRZ have to be made via other MCDU pages and therefore an automatic PERF PAGE change is impossible.

Reply:
@beheh I've improved and updated the code to follow your suggestion.
Only "autochange" performance pages "forward".